tools/python/boutiques/localExec.py

```diff
@@ -435,7 +435,24 @@ def _localExecute(self, command):
             sys.stderr.write('Input Value Error during attempted execution!')
             raise e
         else:
-            return process.communicate(), process.returncode
+            if self.stream:
+                stdoutdata = ""
+                stderrdata = ""
+                while True:
+                    if process.poll() is None:
+                        outLine = process.stdout.readline().decode()
+                        errLine = process.stderr.readline().decode()
+                        if outLine != '':
+                            sys.stdout.write(outLine + "\n")
+                            stdoutdata += outLine
+                        if errLine != '':
+                            sys.stderr.write(errLine + "\n")
+                            stderrdata += errLine
+                    else:
+                        break
+                return (stdoutdata, stderrdata), process.returncode
+            else:
+                return process.communicate(), process.returncode
```

Comment:
Style nitpick: starting with `if not self.stream: return ...` would remove the need for `else` and would save one level of indentation. Likewise, in the `while`, doing `if not process.poll() is None: break` would remove the need for `else`. In general, try to avoid having more than 3 indentation levels (I know, some sections of this file are terrible in this respect).